### ISSUE TITLE
Third Party Service Injections with in-repo addons

### ISSUE THREAD

**Issue by drewcovi** [ember-cli/ember-cli#7620]
Any idea on whether we should be able to use third-party addon services within in-repo addons?

I'm struggling to pull in `ember-cordova/events` into an in-repo addon (shared component for various engines) despite the fact it's available in the host application.

Should this be possible?

It seems services should be available to in-repo addons, but perhaps I'm missing something.

Addon is working great otherwise...

```
import Component from '@ember/component';
import layout from '../templates/components/table-cell-toggle';

export default Component.extend({
  layout,
  cordovaEvents: Ember.inject.service('ember-cordova/events'),
  ...
});
```
---
```
ember-cli: 2.16.2
http_parser: 2.7.0
node: 7.9.0
v8: 5.5.372.43
uv: 1.11.0
zlib: 1.2.11
ares: 1.10.1-DEV
modules: 51
openssl: 1.0.2k
icu: 58.2
unicode: 9.0
cldr: 30.0.3
tz: 2016j
os: darwin x64
5.6.0
1.3.2
```

**Comment by ryanlabouve** [comment]
Do you have it listed as a dependency inside the `lib/{{nameOfYourAddon}}/package.json`? 

e.g.
```
  "dependencies": {
    "ember-cordova-eventsl": "*",
  }
```

**Comment by drewcovi** [comment]
I did. It didn’t seem to matter. Although upon further investigation there’s an open RFC on promoting in-repo addons to first class citizens. 

https://github.com/ember-cli/rfcs/pull/60

**Comment by drewcovi** [comment]
That said this part confuses me. Because I must be getting something wrong. 

```
An in-repo addon has exactly the build-time devDependencies and runtime dependencies as the parent application.
```

**Comment by Turbo87** [comment]
> Should this be possible?

@drewcovi yes, that should be working fine in general.

**Comment by Turbo87** [comment]
I just saw `Ember.inject.service('ember-cordova/events')` and I think the issue here is that you tried to use a sort of namespaced service. Services in Ember only exist in the app itself and `ember-cordova` most likely reexports the `events` service in its `app/services` folder.

tl;dr using `Ember.inject.service('events')` should work

**Comment by drewcovi** [comment]
@Turbo87 still running into huge issues here, specifically with ember-bootstrap, but really any addon fails. is there a dependency on ember version to ensure vendor addons show up in in-repo-addons? any other way to check?
